yajsapi/executor/executor.ts

```diff
@@ -1,57 +1,50 @@
-import { Package } from "../package";
-import { MarketService, MarketStrategy } from "../market";
-import { AgreementPoolService } from "../agreement";
+import { Package, PackageOptions } from "../package";
+import { DemandOptions, MarketService } from "../market";
+import { AgreementOptions, AgreementPoolService } from "../agreement";
 import { Task, TaskQueue, TaskService, Worker } from "../task";
 import { PaymentService } from "../payment";
 import { NetworkService } from "../network";
-import { Result } from "../activity";
+import { ActivityOptions, Result } from "../activity";
 import { sleep, Logger, runtimeContextChecker } from "../utils";
 import { StorageProvider, GftpStorageProvider } from "../storage/";
 import { ExecutorConfig } from "./config";
 import { Events } from "../events";
 import { StatsService } from "../stats/service";
+import { TaskOptions } from "../task/service";
+import { BasePaymentOptions } from "../payment/config";
+import { NetworkServiceOptions } from "../network/service";
+import { AgreementServiceOptions } from "../agreement/service";
+import { WorkOptions } from "../task/work";
+import { LogLevel } from "../utils/logger";
 
 export type ExecutorOptions = {
   /** Image hash as string, otherwise Package object */
   package: string | Package;
-  /** Number of maximum parallel running task on one TaskExecutor instance */
-  maxParallelTasks?: number;
   /** Timeout for execute one task in ms */
-  timeout?: number;
-  /** TODO */
-  budget?: number;
-  /** Strategy used to choose best offer */
-  strategy?: MarketStrategy;
+  executorTimeout?: number;
```

Comment:
`taskTimeout` will be more meaningful. We also need `executorTimeout`vfor all execution timeout